addons/spreadsheet/models/spreadsheet_mixin.py

```diff
@@ -21,6 +23,45 @@ class SpreadsheetMixin(models.AbstractModel):
     spreadsheet_data = fields.Text(compute='_compute_spreadsheet_data', inverse='_inverse_spreadsheet_data')
     thumbnail = fields.Binary()
 
+    @api.constrains("spreadsheet_binary_data")
+    def _check_spreadsheet_data(self):
+        for spreadsheet in self.filtered("spreadsheet_binary_data"):
+            try:
+                data = json.loads(base64.b64decode(spreadsheet.spreadsheet_binary_data).decode())
+            except (json.JSONDecodeError, UnicodeDecodeError):
+                raise ValidationError(_("Uh-oh! Looks like the spreadsheet data file isn't quite right."))
+            if data.get("[Content_Types].xml"):
+                # this is a xlsx file
+                continue
+            display_name = spreadsheet.display_name
+            errors = []
+            for model, field_chains in fields_in_spreadsheet(data).items():
+                if model not in self.env:
+                    errors.append(f"- model '{model}' used in '{display_name}' does not exist")
+                    continue
+                for field_chain in field_chains:
+                    field_model = model
+                    for fname in field_chain.split(
```

Comment:
nitpick but could we have this in a single line?

Reply:
Sure.
Done